### ISSUE TITLE
Improve checkstyle format

### ISSUE THREAD

**Issue by nikolay-bukharev** [eslint/eslint#3183]
Please just add proper source to errors:
lib/formatters/checkstyle.js:58

```javascript
        (message.ruleId ? " (" + message.ruleId + ")" : "") + "\" " +
        "source=\"" + (xmlEscape("bzzz.eslint." + message.ruleId) || '') + "\" />";
```

then checkstyle will parse category (eslint) and type (ruleId) properly.

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by IanVS** [comment]
It seems you have provided a solution, but could you please give us some details about the problem you're having?

**Comment by nikolay-bukharev** [comment]
Checkstyle plugin for Jenkins uses source to find type and category for errors. It parses "source" of error based on dots :) so I just want to have more suitable source for checkstyle plugin.
For instance, you have source="semi":
- with existing code and it means empty type and empty category for us. 
- with my proposal it means type = eslint, category = semi. And you can sort or filter them in Jenkins.

**Comment by nzakas** [comment]
Are these attributes on XML elements? Can you show us an example?

**Comment by nikolay-bukharev** [comment]
```xml
<?xml version="1.0" encoding="utf-8"?>
<checkstyle version="4.3">
<file name="app\js\utils\loglevel.js">
<error line="24" column="2" severity="error" message="Trailing spaces not allowed. (no-trailing-spaces)" source="bzzz.eslint.no-trailing-spaces"/>
<error line="26" column="22" severity="error" message="A function with a name starting with an uppercase letter should only be used as a constructor. (new-cap)" source="bzzz.eslint.new-cap"/>
<error line="27" column="10" severity="warning" message="Strings must use singlequote. (quotes)" source="bzzz.eslint.quotes"/>
<error line="28" column="18" severity="warning" message="Missing semicolon. (semi)" source="bzzz.eslint.semi"/>
</file>
</checkstyle>
```

**Comment by nzakas** [comment]
OK, just so I'm clear, `source` is currently just "semi" and you want it to be "eslint.semi"?

**Comment by nikolay-bukharev** [comment]
bzzz.eslint.semi :)
https://github.com/jenkinsci/checkstyle-plugin/blob/f93909d278db9758647353bfc383c58f06ab3b33/src/main/java/hudson/plugins/checkstyle/parser/CheckStyleParser.java
113

it wants to have structure like <anything>dot<category>dot<type>

**Comment by nzakas** [comment]
What is "bzzz"?

**Comment by nikolay-bukharev** [comment]
Just a placeholder. Please use "io.github" instead, to have full name like  io.github.eslint.semi
or "org", to have org.eslint.semi - don't know what you prefer.

**Comment by nzakas** [comment]
Oh, I see, it has to have three pieces. How about `eslint.rules.semi`?

**Comment by nikolay-bukharev** [comment]
It would work, so I'm fine with this proposal.
